cmd/root/root.go

```diff
@@ -151,5 +161,40 @@ Stack Trace:
 		}
 	}
 
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), startTime, exitCode)
 	return err
 }
+
+func uploadTelemetry(ctx context.Context, cmdStr string, startTime time.Time, exitCode int) {
+	// Return early if there are no logs to upload.
+	if !telemetry.HasLogs(ctx) {
+		log.Debugf(ctx, "no telemetry logs to upload")
+		return
+	}
+
+	// Telemetry is disabled. We don't upload logs.
+	if os.Getenv(telemetry.DisableEnvVar) != "" {
+		log.Debugf(ctx, "telemetry upload is disabled. Not uploading any logs.")
+		return
+	}
+
+	telemetry.SetExecutionContext(ctx, protos.ExecutionContext{
+		CmdExecID:       cmdExecId,
+		Version:         build.GetInfo().Version,
+		Command:         cmdStr,
+		OperatingSystem: runtime.GOOS,
+		DbrVersion:      env.Get(ctx, dbr.EnvVarName),
```

Comment:
Please store the DBR version on the context and keep it localized to `libs/dbr`.

We already pull the version there, so I rather have a function in that package to retrieve it instead of going through the env var again here.

Reply:
Addessed in https://github.com/databricks/cli/pull/2438. Will update this PR once that's merged.

Reply:
Done